### PR TITLE
Document the #[module] macro

### PR DESCRIPTION
**Pull request by dvdplm**

### CODE REVIEW

File: proc-macro/src/lib.rs [hunk 1]
```diff
@@ -30,6 +30,54 @@ use synstructure::{
     Structure,
 };
 
+/// Register type sizes for [EventsDecoder](struct.EventsDecoder.html)
```
Comment (ascjones) [inline proc-macro/src/lib.rs:33]: I think we should also mention the module name constant, which will be the same as the trait ident. This in turn must match the name of the substrate pallet. Indeed the codegen for events and calls depends on this constant, so we might mention that this attribute is required

File: proc-macro/src/lib.rs [hunk 2]
```diff
@@ -30,6 +30,54 @@ use synstructure::{
...
+///
+/// Example:
+///
+/// ```ignore
+/// #[module]
+/// pub trait Herd: Husbandry {
+/// 	type Hooves: HoofCounter;
+/// 	type Wool: WoollyAnimal;
+/// 	#[module(ignore)]
+/// 	type Digestion: EnergyProducer + std::fmt::Debug;
```
Comment (ascjones) [inline proc-macro/src/lib.rs:47]: Indentation

File: proc-macro/src/lib.rs [hunk 3]
```diff
@@ -30,6 +30,54 @@ use synstructure::{
...
+///     // Registers this modules types.
+///     fn with_herd(&mut self);
+/// }
+///
+/// impl<T: Herd> HerdEventsDecoder for
+///     substrate_subxt::EventsDecoder<T>
+/// {
+///     fn with_herd(&mut self) {
+///         self.with_husbandry();
+///         self.register_type_size::<T::Hooves>("Hoves");
```
Comment (ascjones) [inline proc-macro/src/lib.rs:74]:
```suggestion
///         self.register_type_size::<T::Hooves>("Hooves");
```

File: proc-macro/src/lib.rs [hunk 4]
```diff
@@ -30,6 +30,54 @@ use synstructure::{
...
+///         self.register_type_size::<T::Wool>("Wool");
+///     }
+/// }
+/// ```
+///
+/// The size of the following types is registered by default: `bool, u8, u32, AccountId, AccountIndex, AuthorityId, AuthorityIndex, AuthorityWeight, BlockNumber, DispatchInfo, Hash, Kind, MemberCount, PhantomData, PropIndex, ProposalIndex, ReferendumIndex, SessionIndex, VoteThreshold`
```
Comment (ascjones) [inline proc-macro/src/lib.rs:80]:
```suggestion
/// The following type sizes are registered by default: `bool, u8, u32, AccountId, AccountIndex, AuthorityId, AuthorityIndex, AuthorityWeight, BlockNumber, DispatchInfo, Hash, Kind, MemberCount, PhantomData, PropIndex, ProposalIndex, ReferendumIndex, SessionIndex, VoteThreshold`
```

File: proc-macro/src/module.rs [hunk 5]
```diff
@@ -69,7 +69,7 @@ fn events_decoder_trait_name(module: &syn::Ident) -> syn::Ident {
 fn with_module_ident(module: &syn::Ident) -> syn::Ident {
     format_ident!("with_{}", module.to_string().to_snake_case())
 }
-
+/// Attribute macro that registers the type sizes used by the module.
```
Comment (ascjones) [inline proc-macro/src/module.rs:72]: Might want to mention about the module name const which is required for the other derive macros to work

File: proc-macro/src/lib.rs (hunk 2 above)
Comment (dvdplm) [reply proc-macro/src/lib.rs:47]: Sorry, I don't see it, what is not right here?

Comment (ascjones) [reply proc-macro/src/lib.rs:48]: Looks like 1 space instead of 4
image

Comment (dvdplm) [reply proc-macro/src/lib.rs]: Oh, we use spaces here, good catch. Odd that my github shows me the right white space (as did my editor, but you're right, I had one space and one tab here).

**Pull request merged**